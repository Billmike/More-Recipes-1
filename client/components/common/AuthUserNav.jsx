import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * User navigation when unathenticated
 *
 * @param {any} props 
 * @returns {object} React element
 */
const AuthUserNav = (props) => {
  const { handleLogoutUser } = props;
  return (
    <ul id="user-control" className="dropdown-content">
      <li>
        <a href="profile.html" className="black-text">
          Profile
          <i className="material-icons left">person</i>
        </a>
      </li>
      <li>
        <Link to="/user/recipes" className="black-text">
          My Recipes<i className="material-icons left">folder</i>
        </Link>
      </li>
      <li>
        <a
          href="recipes.html?#user-favorites"
          className="black-text"
        >
          My Favorites
          <i className="material-icons left">folder</i>
        </a>
      </li>
      <li className="divider" />
      <li>
        <a
          role="button"
          tabIndex="0"
          className="black-text"
          onClick={handleLogoutUser}
        >
          Logout
          <i className="material-icons left">lock</i>
        </a>
      </li>
    </ul>
  );
};

AuthUserNav.propTypes = {
  handleLogoutUser: PropTypes.func.isRequired,
};

export default AuthUserNav;
