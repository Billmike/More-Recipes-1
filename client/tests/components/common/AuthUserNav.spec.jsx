import React from 'react';

import AuthUserNav from '../../../components/common/AuthUserNav';
import userMockData from '../../__mocks__/user';

describe('Authenticated user navigation component', () => {
  beforeEach(() => {
    localStorage.setItem(
      'token',
      userMockData.signinSuccessResponse.data.user.token
    );
  });
  const props = {
    handleLogoutUser: jest.fn(),
  };
  it('should render component', () => {
    const wrapper = shallow(<AuthUserNav {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle user logout onclick action', () => {
    const wrapper = shallow(<AuthUserNav {...props} />);
    wrapper.find('#logout').simulate('click');
    expect(props.handleLogoutUser).toBeCalled();
  });
});
