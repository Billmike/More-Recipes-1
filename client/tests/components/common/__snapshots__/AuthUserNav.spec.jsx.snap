// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticated user navigation component should render component 1`] = `
<ul
  className="dropdown-content"
  id="user-control"
>
  <li
    id="dashboard"
  >
    <Link
      className="black-text"
      replace={false}
      to="/users/2/dashboard"
    >
      Dashboard
      <i
        className="material-icons left"
      >
        person
      </i>
    </Link>
  </li>
  <li
    className="divider"
  />
  <li>
    <a
      className="black-text"
      id="logout"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Logout
      <i
        className="material-icons left"
      >
        lock
      </i>
    </a>
  </li>
</ul>
`;
